### ISSUE TITLE
Source Code Pro isn't monospaced the same as Courier New / other monospaced fonts

### ISSUE THREAD

**Issue by SleeplessByte** [exercism/exercism#6225]
For some reason, monospaced font on Exercism is displaying our (important) ASCII Art incorrectly. This may be related to https://github.com/adobe-fonts/source-code-pro/issues/261, although I find it more likely that we chose to subset this font? 

Anyway, it's broken at the moment. Resolutions:
- Make it work with Source Code Pro, if can be determined that we don't use unsupported characters
- Allow us to use a system monospaced font

## Source Code Pro
![Source Code Pro on exercism showing ascii art incorrectly](https://user-images.githubusercontent.com/1964376/152718826-53d257be-f6a3-421d-b55a-c8e627507056.png)

## monospace system font
![monospace system font on exercism showing ascii art correctly](https://user-images.githubusercontent.com/1964376/152718693-d49a13d4-12a0-40a8-ae7a-e771bdec73d0.png)

## Courier New
![Courier New font on exercism showing ascii art correctly](https://user-images.githubusercontent.com/1964376/152719099-6a5a105d-a7e5-402a-8c45-feafa2ce3526.png)

## Desktop (Mona Lisa and some mono system font)
![Mona Lisa and courier new on desktop showing ascii art correctly](https://user-images.githubusercontent.com/1964376/152718879-425e7357-ff3d-433f-8a60-bb6ea0d7d224.png)

**Comment by iHiD** [comment]
More info. This the font we load: https://fonts.googleapis.com/css2?family=Poppins:ital,wght@0,100;0,200;0,300;0,400;0,500;0,600;0,700;0,800;0,900;1,100;1,200;1,300;1,400;1,500;1,600;1,700;1,800;1,900&family=Source+Code+Pro:ital,wght@0,200;0,300;0,400;0,500;0,600;0,700;0,900;1,200;1,300;1,400;1,500;1,600;1,700;1,900&display=swap

If I can amend that to fix, tell me how and I'll do it.

Do you know if any of the chars in the issue you linked to are in the text ascii art?

**Comment by angelikatyborska** [comment]
I did some debugging by manually manipulating the DOM to insert the ASCII art all wrapped in span to produce this:
![Screenshot 2022-02-07 at 18 22 34](https://user-images.githubusercontent.com/7852553/152839530-332c67ce-fa13-4fc1-9840-2ab1d7ad8548.png)

It looks to me like those characters are too wide:
```
═
─
```

Note to self, In case I want to check it again, here's the debug DOM that I randomly inserted in the page:
```
<pre id="foo"><code><span></span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>&lt;</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span> </span><span>s</span><span>c</span><span>r</span><span>e</span><span>e</span><span>n</span><span>S</span><span>i</span><span>z</span><span>e</span><span>.</span><span>w</span><span>i</span><span>d</span><span>t</span><span>h</span><span> </span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>&gt;</span><span>
</span><span>
</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>^</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>╔</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>╗</span><span>
</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>|</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span>
</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>|</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>p</span><span>o</span><span>s</span><span>i</span><span>t</span><span>i</span><span>o</span><span>n</span><span>.</span><span>x</span><span>,</span><span>_</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span>
</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>|</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>p</span><span>o</span><span>s</span><span>i</span><span>t</span><span>i</span><span>o</span><span>n</span><span>.</span><span>y</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span>
</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>|</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>&lt;</span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span> </span><span>s</span><span>i</span><span>z</span><span>e</span><span>.</span><span>w</span><span>i</span><span>d</span><span>t</span><span>h</span><span> </span><span>-</span><span>-</span><span>-</span><span>-</span><span>-</span><span>&gt;</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span>
</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>|</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>^</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>*</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>┐</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span>
</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>|</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>|</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>│</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>t</span><span>i</span><span>t</span><span>l</span><span>e</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>│</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span>
</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>|</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>|</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>├</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>┤</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span>
</span><span>s</span><span>c</span><span>r</span><span>e</span><span>e</span><span>n</span><span>S</span><span>i</span><span>z</span><span>e</span><span>.</span><span>h</span><span>e</span><span>i</span><span>g</span><span>h</span><span>t</span><span> </span><span>║</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>|</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>│</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>│</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span>
</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>|</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>s</span><span>i</span><span>z</span><span>e</span><span>.</span><span>h</span><span>e</span><span>i</span><span>g</span><span>h</span><span>t</span><span> </span><span>│</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>│</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span>
</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>|</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>|</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>│</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>c</span><span>o</span><span>n</span><span>t</span><span>e</span><span>n</span><span>t</span><span>s</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>│</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span>
</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>|</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>|</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>│</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>│</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span>
</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>|</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>|</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>│</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>│</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span>
</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>|</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>v</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>└</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>─</span><span>┘</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span>
</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>|</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span>
</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>|</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>║</span><span>
</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>v</span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span> </span><span>╚</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>═</span><span>╝</span><span>
</span><span></span>
</code>
</pre>
```

**Comment by angelikatyborska** [comment]
Calculating further, those character codes are:
```
dec 9472 = hex 2500
dec 9552 = hex 2550
```

I think they are not included in the unicode ranges in the loaded font but I'm not 100% because I'm just eyeing this. E.g. `latin-ext`:
```
  unicode-range: U+0100-024F, U+0259, U+1E00-1EFF, U+2020, U+20A0-20AB, U+20AD-20CF, U+2113, U+2C60-2C7F, U+A720-A7FF;
```

**Comment by SleeplessByte** [comment]
What @angelikatyborska is saying seems correct to me @iHiD . Adding those codes would fix it!

**Comment by iHiD** [comment]
Amazing work - thanks @angelikatyborska.

> Adding those codes would fix it!

Do either of you know how to do this? :)

**Comment by angelikatyborska** [comment]
> > Adding those codes would fix it!

> Do either of you know how to do this? :)

This is a guess: I think it's not possible from Google fonts (without self hosting). With self hosting, you would need to find a file for this font that actually implements those glyphs. Some font editing app can be used to inspect a font file in this regard. I wouldn't bet on this font implementing those fancy glyphs at all.

TL;DR: I would say we should either: just remove the custom font and only use the system default `mono` font face, or redraw the ASCII boxes using the basic characters `-` and `_` instead.

**Comment by joshgoebel** [comment]
> redraw the ASCII boxes using the basic characters - and _ instead.

Is it really ASCII art at all if it's using extended UTF-8 characters? ;-)  If this is a systemic problem I wonder if we couldn't do search/replace in software during the render process vs changing ALL the content... (I don't know how pervasive this is)

**Comment by SleeplessByte** [comment]
I'll think about it for a minute. I definitely would expect `-` and `=` to work, but maybe it's not regular `-` and regular `=`.
